IATI/refresher
# Next Release

- validate validating schema first then full validation
- remove adhoc validation code as this was replaced with validator-services functions
- remove unused dep
- add safety valve logic into validate to delay validation of schema-invalid docs
- clear file_schema_valid when an existing document is updated
- move safety check to validate library and prevent full validation for schema-invalid files if publisher is black flagged
- formatting
- remove unused column and variable
- rename and format
- rename alv to clean
- close connection and log that we're finished for safety check
- update options
- basics of copying valid activities docs to clean container
- format
- add document_id blob index tag to clean blob
- ensure clean container is cleaned up for stale and changed docs
- refactor alv container to clean container
- separate cleaning from source and clean error handling, update logging levels
- remove unused dependencies
- remove errant comment
- update flatten and lakify to pull documents from clean container
- don't validate files with blank hashes
- drop alv_relvalidate column as its now unnecessary
- increment version to drop alv_relvalidate column as its now unnecessary
- update migrations to trigger re-validation of files with alv2 changes
- deployment changes for alv 2
- only add meta=true for invalid docs
- don't trigger revalidation of non-alv docs in migration
- activity-level to -clean for tagging
- add clean container name env var to refresh container
- formatting
- get flattened activities by id instead of hash
- add black flag check before download as well
- Editorconfig 4 spaces
- Update extensions
- Transform location_point_pos for SOLR LatLonPointSpatialField
- Always append
- Change name
- Add validated coordinates before adding activity to solr, and fix for str types
- increase pysolr timeout and reduce batch length to prevent timeouts
- reduce requested CPU for containers based on usage
- reduce requested CPU for containers based on usage
- reduce requested memory for containers based on usage
- update ACU to match cpu=0.1
- reduce number of parallel processes for clean to prevent crashes
- increase memory for flattener after crashes
- increase validate memory after crashes
- delete payload after request to free memory
- increase memory for lakify and flatten
- increase memory on clean
- one container in yaml
- few more updates to deployment test for 1 container
- try deploying refresh with yaml in single group
- fix env var definition
- Revert "fix env var definition"
- Revert "Merge pull request #218 from IATI/tci/multi-container-group"
- Revert "Merge pull request #219 from IATI/revert-multicont"
- simplify container name
- change delimiter to ^ for sed command
- Revert "Merge pull request #220 from IATI/tci/multi-container-group"
- add document.name column to store registry dataset name
- sync document name from the registry
- remove brackets from templating
- make env var from image name
- deployment modifications
- debug cat
- try direct env var instead of github format
- update yml formatting, add scratch to gitignore
- deploy validate in main container group
- fix indentation of resource requests
- delete new container group before deploying
- add clean to the container group
- fix environment varaible in clean deployment
- add flatten to the container group
- add lakify to the container group
- add solrize to the container group
- remove unused env vars
- put non-secret env vars at top for easier access
- refactoring and cleanup
- increase cpu on refresh due to slow downloading
- set ACU tag to correct value after upping refresher CPU
- document deployment info
- fix tag values to match others
- fix tags
- increase clean memory due to crashes
- Recursive nest activity XML into JSON
- Solrize into iati_json field
- Encode JSON utf-8
- Include blank strings for narratives
- fix clean entrypoints in handler
- documentation and environment variable structure updates
- Remove Xpath syntax
- Revert "Remove Xpath syntax"
- Handle XML comments and PIs
- Test more solrize memory
- formatting
- formatting
- add back SOLR_PARALLEL_PROCESSES into deployment and bump to 5
- update to new deployment method
- remove unused env var
- use name env var in delete
- match dev version for now
